### PR TITLE
async_std support through feature flags

### PR DESCRIPTION
**Pull request by schulke-214**

### CODE REVIEW

**Comment by schulke-214** [comment]
Still a draft, i need to migrate tls support aswell

**Comment by Jarema** [comment]
@schulke-214 for ASAP cases `compat` mode of `async-std` works really well - performance-wise there is no noticeable difference between `async-std` and native `tokio`, just small memory overhead.

That's a very interesting PR though!

**Comment by schulke-214** [comment]
If i manage to Port everything and pass the Tests would you accept it?

Compat Mode sounds interesting but having no tokio dependency in async std environments is still cleaner imo.

**Comment by caspervonb** [comment]
Hey @schulke-214, thanks for the PR!

I'm interested in using neutral types as more traits are popping up in the futures crate at-least, but things might break a lot (as it has already for this PR) as we're moving stuff around and doing feature work.

Public interfaces won't change (much at-least) but merge conflicts are going to be a bit hectic in the coming days so I'd advice against doing too much as it'll be tedious to resolve the merge conflicts.

We'll be adding some benchmarks very soon so we can test this what kind of impact this sort of thing has on footprint and performance with some degree of confidence.

TL;DR: Give us a couple of days 😄

**Comment by schulke-214** [comment]
No problem :) Thanks for the update! If you need anything let me know :slightly_smiling_face:

**Comment by mara214** [comment]
Okay thank you! :)
